### ISSUE TITLE
docker image via nix

### ISSUE THREAD

**Issue by dckc** [monte-language/typhon#77]

**Comment by MostAwesomeDude** [comment]
The `docker-nix` tool is retired in favor of some sort of `nixos-rebuild`-powered thing.

While we're at it, I should figure out how to do Ubuntu-compatible packaging, for those who don't have Docker.

**Comment by dckc** [comment]
Ubuntu-compatible packaging seems like a whole other ball of wax. Monte and nix are pretty intertwingled at this point.

**Comment by MostAwesomeDude** [comment]
I made a Docker tarball by doing this to an ``airbrus`` repo:

    $ mt build
    $ nix-repl -f ~/typhon/default.nix
    nix-repl> :b dockerize { lockfile = ./mt-lock.json; name = "airbrus"; }

This obviously isn't production-ready but all the pieces are there.

Also I have no idea how to Docker, so I don't know how to test that the tarball works.

**Comment by dckc** [comment]
I know how to Docker a little bit, but I'm losing when I try to make an image:

```
nix-repl> :b dockerize { lockfile = ./mt-lock.json; name = "mt-mafia"; }
trace: mafiabot
these derivations will be built:
  /nix/store/vbynjzp50z2dzi96xiwvmwyflnh509jn-mt-mafia.drv
  /nix/store/3c1jwprh33a50z0rng2blp246m26bmpb-docker-layer.drv
  /nix/store/zszni54j08aa864j3831523bpzcqiqf0-runtime-deps.drv
  /nix/store/bvr6l179m3jzn2fzsqg9csi93khy45p8-mt-mafia-deps.tar.gz.drv
  /nix/store/b67kjcgciggsifdzdxn41swx0vi5dryj-mt-mafia.tar.gz.drv
building path(s) ‘/nix/store/1fg6psw18s2072k1y688098y5c0cabqc-mt-mafia’
unpacking sources
unpacking source archive /nix/store/pyzw86vnr1fn930x0w7w5h5l8c1ap9zg-mt-mafia
source root is mt-mafia
patching sources
configuring
no configure script, doing nothing
building
Taking initial turn in script...
TRACE: ["audit/1: Missing methods: [[\"progressed\", 1]].asSet()"]
TRACE: ["<readInputFile> took 0.071460 seconds"]
TRACE: ["audit/1: Missing methods: [[\"started\", 0], [\"progressed\", 1], [\"stopped\", 1]].asSet()"]
TRACE: ["audit/1: Missing methods: [[\"started\", 0], [\"received\", 1], [\"progressed\", 1], [\"stopped\", 1]].asSet()"]
Log: (vat:serious) Vat pa: Uncaught user exception while taking turn (and no resolver): Exception in user code:
In <pumpTube>.Atom(abortFlow/0) []:
In <FileDrain>.Atom(flowAborted/0) []:
Message refused: (<FileDrain>, Atom(flowAborted/0), [])
Problem: Name usage error
In throw.Atom(run/1) ["Name usage error", "Name usage error"]:
In <parse>.Atom(run/1) [import "mafia" =~ [=> makeMafia :DeepFrozen]
import "irc/client" =~ [=> makeIRCClient :DeepFrozen,
                        => connectIRCClient :DeepFrozen]
...
```

p.s. no -f on the nix-repl command

**Comment by washort** [comment]
We now have a `mtDocker` nix attribute; this produces a repl-only docker image (including 'mt build' etc pulls in gcc, git, and a host of other things). I've uploaded one to Docker Hub as `montelang/repl`.

Docker images can be built from Monte packages using `mt docker-build`.

(To run a freshly built docker image: run `docker load < result` then `docker run -it [imgname]`.

**Comment by dckc** [comment]
yup. fixed in 68563d64c55bc91cbdfc463470799d5a2bc3bcb1

The https://hub.docker.com/r/montelang/repl/ entry could be prettied up, but it works.

```
$ docker pull montelang/repl
Using default tag: latest
latest: Pulling from montelang/repl
f96cfaec20e1: Already exists 
Digest: sha256:5600d4c85f086465063e3598768cc0a62d27897ad8af44e7d10804bb2bcbaef8
Status: Image is up to date for montelang/repl:latest
```

**Comment by dckc** [comment]
I'm having less luck with `mt docker-build`, but I regard that as separate from this issue.
